src/PIL/ImageTk.py

```diff
@@ -190,10 +181,10 @@ def paste(self, im: Image.Image) -> None:
         if image.isblock() and im.mode == self.__mode:
             block = image
         else:
-            block = image.new_block(self.__mode, im.size)
+            block = Image.core.new_block(self.__mode, im.size)
```

Comment:
I imagine this change was just because you didn't like `image.new_block(self.__mode, im.size)`?

Rather than moving `new_block` in C, wouldn't it have been better for the C method to access the size on the image object instead of being passed it? Then this call could become `image.new_block(self.__mode)`.

Reply:
I’d like to clarify that the reason for this change is not a matter of personal preference. The `new_block` method doesn't pertain to an existing image but instead creates a new block based on the passed parameters. As such, it makes more sense for it to reside in the `Image.core` module rather than being a method of the image object itself.

There is no intent to modify or enhance the behavior; I was simply correcting the method's improper placement.